src/sentry/static/sentry/app/views/settings/components/settingsProjectItem.jsx

```diff
@@ -33,40 +26,23 @@ const ProjectItem = createReactClass({
     };
   },
 
-  handleToggleBookmark() {
-    const {project, organization} = this.props;
-    const {isBookmarked} = this.state;
-
-    this.setState({isBookmarked: !isBookmarked}, () =>
-      update(this.api, {
-        orgId: organization.slug,
-        projectId: project.slug,
-        data: {
-          isBookmarked: this.state.isBookmarked,
-        },
-      })
-    );
-    //needed to dismiss tooltip
-    document.activeElement.blur();
+  handleToggleBookmark(isBookmarked) {
+    this.setState({isBookmarked});
   },
 
   render() {
     const {project, organization} = this.props;
-    const {isBookmarked} = this.state;
 
     const hasNewRoutes = new Set(organization.features).has('sentry10');
 
     return (
-      <div key={project.id} className={isBookmarked ? 'isBookmarked' : null}>
-        <Tooltip title={isBookmarked ? 'Remove from bookmarks' : 'Add to bookmarks'}>
-          <InlineButton onClick={() => this.handleToggleBookmark()}>
-            {isBookmarked ? (
-              <span className="icon-star-solid bookmark" />
-            ) : (
-              <span className="icon-star-outline bookmark" />
-            )}
-          </InlineButton>
-        </Tooltip>
+      <Container key={project.id}>
+        <BookmarkLink
+          organization={organization}
+          project={project}
+          isBookmarked={this.state.isBookmarked}
```

Comment:
Is this to work around the `project` not having its `isBookmarked` property updated after being bookmarked? Is that something we could do in the BookmarkLink component after successfully making the API call?

Reply:
yeah—I wasn't sure how to update `ProjectsStore` after the user made a text search. We can store the internal state in `BookmarkStar` but that state is lost if the component is unmounted for any reason.

Reply:
You can use `ProjectsStore.onUpdateSuccess({id: project.id, isBookmarked: !isBookmarked})` to update the store after the update has completed.

Reply:
ok I tried a bunch of stuff here, but i'm running into a lot of wackiness. we can store the `isBookmarked` state in the star component itself after a click, or even on a successful server response, but then I run into source-of-truth headaches around whether the internal state or new incoming props are correct. Take for example:

1. I click "star" on a project card. it stars it, as well as the version in the dropdown header
2. I then go and click "unstar" in the dropdown header
3. The project card receives an update upstream, but which `isBookmarked` is correct? The internal state or the external props? What if someone changes another feature that has nothing to do with the bookmarking? Who do we trust? Who built the pyramids? Are we really alone in this vast universe?

Anyways—you get the point. My intuition is that we need something that is operated by a controller with a state _or_ a high-order data store, but we probably shouldn't mix the two. Maybe we could chat real time tomorrow about this problem, but I'm gonna leave it as it is for now because it just works™

Reply:
Having both internal state and props would get trouble some. We can move forward with this and figure out how to tie in store state later.